### PR TITLE
TodoList : Implement storage system

### LINKED ISSUE #179: TodoList: Implement storage to store the list

### PR DESCRIPTION
**Pull request by davidcaiqifan**
Fixes #179 
- Adds mutiple classes to faciliate storage of todolist
- Integrates storage with rest of code

### CODE REVIEW

File: src/main/java/seedu/address/storage/StorageManager.java [hunk 1]
```diff
@@ -108,4 +111,32 @@ public void saveContactList(ReadOnlyContactList contactList, Path filePath) thro
...
+    }
+
+    @Override
+    public void saveTodoList(ReadOnlyTodoList todoList) throws IOException {
+        saveTodoList(todoList, todoListStorage.getTodoListFilePath());
+    }
+
+    @Override
+    public void saveTodoList(ReadOnlyTodoList todoList, Path filePath) throws IOException {
+        logger.fine("Attempting to write to data file: " + filePath);
```
Comment (jonasngs) [inline src/main/java/seedu/address/storage/StorageManager.java:139]: The comment from above could be useful here as well, we could refine the logger message to be something more specific and meaningful.

File: src/main/java/seedu/address/storage/JsonAdaptedTask.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        this.tag = tag;
+        this.priority = priority;
+        this.date = date;
+        this.status = status;
+    }
+
+    /**
+     * Converts a given {@code Task} into this class for Jackson use.
+     */
+    public JsonAdaptedTask(Task source) {
```
Comment (jonasngs) [inline src/main/java/seedu/address/storage/JsonAdaptedTask.java:45]: I think the implementation of this constructor can be improved. For a Task object, the Date, Priority and Tag fields are optional and the below implementation might result in a NullPointerException, for eg, invoking getPriority().get() might return null.

**Review by jonasngs — CHANGES_REQUESTED** [review]
The code quality is good, the code is readable and understandable and conforms to the implementation of Module Storage and Contact Storage. It LGTM, albeit some changes needed.

File: src/main/java/seedu/address/storage/JsonAdaptedToDoList.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package seedu.address.storage;
```
[the remark this answers is not in the document]
Comment (davidcaiqifan) [reply src/main/java/seedu/address/storage/JsonAdaptedToDoList.java:1]: It was unnecessary. Thanks for pointing that out!

**Review by jonasngs — APPROVED** [review]

**Comment by jonasngs** [comment]
LGTM.

**Pull request merged**